lib/base-compiler.js

```diff
@@ -895,7 +957,157 @@ export class BaseCompiler {
         buildResult.dirPath = undefined;
     }
 
-    async compile(source, options, backendOptions, filters, bypassCache, tools, executionParameters, libraries) {
+    getCompilerEnvironmentVariables(compilerflags) {
+        if (this.lang.id === 'c++') {
+            return _.extend(_.clone(this.cmakeBaseEnv), {
```

Comment:
Do we need to use underscore here? I think `{...this.cmakeBaseEnv, CXXFLAGS: compilerflags}`  is equivalent here?

```js
> a = {a: 1, b: 2}
< {a: 1, b: 2}
> b = {...a, c:3 }
< {a: 1, b: 2, c: 3}
> a
{a: 1, b: 2}
```